### PR TITLE
Change to the Azure code generator to pass settings

### PR DESCRIPTION
**Pull request by begoldsm**
This enables UseDateTimeOffset to work properly for Azure generated C#
clients which is part of issue: https://github.com/Azure/autorest/issues/825

### CODE REVIEW

File: AutoRest/Generators/CSharp/Azure.CSharp/AzureCSharpCodeGenerator.cs [hunk 1]
```diff
@@ -25,7 +25,8 @@ public class AzureCSharpCodeGenerator : CSharpCodeGenerator
 
         public AzureCSharpCodeGenerator(Settings settings) : base(settings)
         {
-            _namer = new AzureCSharpCodeNamer();
+
```
Comment (amarzavery) [inline AutoRest/Generators/CSharp/Azure.CSharp/AzureCSharpCodeGenerator.cs]: Why an empty line

Comment (begoldsm) [inline AutoRest/Generators/CSharp/Azure.CSharp/AzureCSharpCodeGenerator.cs]: Fixed!

File: AutoRest/Generators/CSharp/Azure.CSharp/AzureCSharpCodeNamer.cs [hunk 2]
```diff
@@ -24,6 +24,28 @@ public class AzureCSharpCodeNamer : CSharpCodeNamer
             // Do nothing
         }
 
+        public AzureCSharpCodeNamer(Settings settings)
+        {
+            if (settings != null && settings.CustomSettings != null)
```
Comment (brodyberg) [inline AutoRest/Generators/CSharp/Azure.CSharp/AzureCSharpCodeNamer.cs]: Is this an error condition for your object? Is it interesting to log about whether or not AzureCSharpCodeNamer got a valid settings object and/or took action on the CustomSetting?

Comment (begoldsm) [inline AutoRest/Generators/CSharp/Azure.CSharp/AzureCSharpCodeNamer.cs]: This is to fix the bug where the base class' property: UseDateTimeOffset is not being set when the AzureCSharpCodeNamer is instantiated. We don't care if settings are passed in, so this constructor now behaves exactly like the default constructor if settings is null, however if settings is not null we will attempt to properly set this value.

**Comment by jianghaolu** [comment]
Hey @azuresdkci retest this please

**Pull request merged**